http/sys_raft.go

```diff
@@ -27,7 +31,40 @@ func handleSysRaftJoinPost(core *vault.Core, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	joined, err := core.JoinRaftCluster(context.Background(), req.LeaderAddr, nil, req.Retry)
+	var tlsConfig *tls.Config
+	switch {
+	case req.LeaderCACert != "" && req.LeaderClientCert != "" && req.LeaderClientKey != "":
+		// Create TLS Config
+		pool := x509.NewCertPool()
```

Comment:
I approved this but if you can move this section to sdk/helper/tlsutil we can reuse it elsewhere pretty quickly...

Reply:
I'll do this in a separate PR.